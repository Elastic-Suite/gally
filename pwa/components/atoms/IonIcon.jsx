import React from 'react'

/* Creation of special props to be clean on Typescripts files
 * See: https://github.com/ionic-team/ionicons
 */
const IonIcon = (props) => {
    return <ion-icon {...props} />
}

export default IonIcon
