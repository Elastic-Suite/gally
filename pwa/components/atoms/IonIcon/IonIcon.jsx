import React from 'react'
import home2 from '~/assets/images/home2.svg'

export const customIcons = [
    'dashboard',
    'analyze',
    'merchandize',
    'monitoring',
    'settings',
]

/*
 * Creation of special props to be clean on Typescripts files
 * See: https://github.com/ionic-team/ionicons
 *
 * Setup switch for special names that need svg or have another name in ion-icons
 */
const IonIcon = (props) => {
    const propsCleaned = { ...props }
    delete propsCleaned.name
    switch (props.name) {
        case 'dashboard':
            return <ion-icon src={home2.src} {...propsCleaned} />
        case 'analyze':
            return <ion-icon name={'analytics'} {...propsCleaned} />
        case 'merchandize':
            return <ion-icon name={'funnel-outline'} {...propsCleaned} />
        case 'monitoring':
            return <ion-icon name={'list'} {...propsCleaned} />
        case 'settings':
            return <ion-icon name={'settings-outline'} {...propsCleaned} />
        default:
            return <ion-icon name={props.name} {...propsCleaned} />
    }
}

export default IonIcon
