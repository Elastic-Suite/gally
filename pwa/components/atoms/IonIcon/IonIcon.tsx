/* eslint-disable @typescript-eslint/no-namespace */
import { CSSProperties, MouseEvent } from 'react'
import { JSX as IonIconJSX } from 'ionicons'
import { JSXBase } from 'ionicons/dist/types/stencil-public-runtime'

export type IIonIconProps = Omit<
  IonIconJSX.IonIcon & JSXBase.HTMLAttributes<HTMLIonIconElement>,
  'style' | 'onClick'
> & {
  style?: CSSProperties
  onClick?: (event: MouseEvent<IIonIconProps>) => void
}

declare global {
  namespace JSX {
    // eslint-disable-next-line @typescript-eslint/naming-convention
    interface IntrinsicElements {
      'ion-icon': IIonIconProps
    }
  }
}

interface IProps extends IIonIconProps {
  name: string
  tooltip?: boolean
}

export const customIcons = [
  'dashboard',
  'analyze',
  'merchandize',
  'monitoring',
  'settings',
]

/*
 * Creation of special props to be clean on Typescripts files
 * See: https://github.com/ionic-team/ionicons
 *
 * Setup switch for special names that need svg or have another name in ion-icons
 */
function IonIcon(props: IProps): JSX.Element {
  const { name, tooltip, ...iconProps } = props
  switch (name) {
    case 'dashboard':
      return <ion-icon src="/images/home2.svg" {...iconProps} />
    case 'analyze':
      return <ion-icon name="analytics" {...iconProps} />
    case 'merchandize':
      return <ion-icon name="funnel-outline" {...iconProps} />
    case 'monitoring':
      return <ion-icon name="list" {...iconProps} />
    case 'settings':
      return <ion-icon name="settings-outline" {...iconProps} />
    case 'information':
      return (
        <ion-icon
          name="information-circle-outline"
          style={
            tooltip
              ? {
                  color: '#8187B9',
                  width: '20px',
                  height: '20px',
                  display: 'flex',
                }
              : { ...iconProps }
          }
        />
      )
    default:
      return <ion-icon name={name} {...iconProps} />
  }
}

export default IonIcon
