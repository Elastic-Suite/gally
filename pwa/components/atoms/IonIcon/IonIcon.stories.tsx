import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { icons } from '~/constants';

import IonIcon from './IonIcon';

export default {
  title: 'Atoms/IonIcon',
  component: IonIcon,
  argTypes: {
    name: {
      options: icons,
      control: { type: 'select' },
    },
  }
} as ComponentMeta<typeof IonIcon>;

const Template: ComponentStory<typeof IonIcon> = (args) => <IonIcon {...args} />;

export const Default = Template.bind({});
Default.args = {
  name: 'analytics'
};
