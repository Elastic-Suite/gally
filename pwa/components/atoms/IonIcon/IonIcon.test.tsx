import { render } from '@testing-library/react'
import IonIcon from './IonIcon'

describe('IonIcon', () => {
  it('render properly dashboard', () => {
    const { container } = render(<IonIcon name="dashboard" />)
    expect(container).toBeTruthy()
  })
  it('render properly analyze', () => {
    const { container } = render(<IonIcon name="analyze" />)
    expect(container).toBeTruthy()
  })
  it('render properly merchandize', () => {
    const { container } = render(<IonIcon name="merchandize" />)
    expect(container).toBeTruthy()
  })
  it('render properly monitoring', () => {
    const { container } = render(<IonIcon name="monitoring" />)
    expect(container).toBeTruthy()
  })
  it('render properly settings', () => {
    const { container } = render(<IonIcon name="settings" />)
    expect(container).toBeTruthy()
  })

  it('match snapshot', () => {
    const { container } = render(<IonIcon name="home" />)
    expect(container).toMatchSnapshot()
  })
})
