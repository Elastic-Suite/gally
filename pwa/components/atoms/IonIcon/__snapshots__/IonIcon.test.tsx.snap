// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IonIcon render properly analyze & match snapshot 1`] = `
<div>
  <ion-icon
    name="analytics"
  />
</div>
`;

exports[`IonIcon render properly dashboard & match snapshot 1`] = `
<div>
  <ion-icon
    src="/images/home2.svg"
  />
</div>
`;

exports[`IonIcon render properly home & match snapshot 1`] = `
<div>
  <ion-icon
    name="home"
  />
</div>
`;

exports[`IonIcon render properly information & match snapshot 1`] = `
<div>
  <ion-icon
    name="information-circle-outline"
  />
</div>
`;

exports[`IonIcon render properly information with tooltip & match snapshot 1`] = `
<div>
  <ion-icon
    name="information-circle-outline"
  />
</div>
`;

exports[`IonIcon render properly merchandize & match snapshot 1`] = `
<div>
  <ion-icon
    name="funnel-outline"
  />
</div>
`;

exports[`IonIcon render properly monitoring & match snapshot 1`] = `
<div>
  <ion-icon
    name="list"
  />
</div>
`;

exports[`IonIcon render properly settings & match snapshot 1`] = `
<div>
  <ion-icon
    name="settings-outline"
  />
</div>
`;
